tools/packaging/helm-chart/kata-deploy/templates/kata-gpu-feature-discovery.yaml

```diff
@@ -0,0 +1,75 @@
+{{- if .Values.kataGpuFeatureDiscovery -}}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kata-gpu-feature-discovery
+  namespace: {{ .Release.Namespace }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: kata-gpu-feature-discovery
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: kata-gpu-feature-discovery
+    spec:
+      runtimeClassName: kata-qemu-nvidia-gpu
+      nodeSelector:
+        katacontainers.io/gpu-feature-discovery: required
+{{- with .Values.kataDeploy.imagePullSecrets }}
+      imagePullSecrets:
+{{- toYaml . | nindent 6 }}
+{{- end }}
+      initContainers:
+        - image: nvcr.io/nvidia/k8s-device-plugin:v0.17.0
+          name: gpu-feature-discovery
+          command: ["/usr/bin/gpu-feature-discovery"]
+          resources:
+            requests:
+              "nvidia.com/pgpu": 1
+            limits:
+              "nvidia.com/pgpu": 1
+          volumeMounts:
+            - name: output-dir
+              mountPath: "/etc/kubernetes/node-feature-discovery/features.d"
+            - name: host-sys
+              mountPath: "/sys"
+          securityContext:
+            privileged: true
+          env:
+            - name: MIG_STRATEGY
+              value: none
+            - name: GFD_ONESHOT
+              value: "true"
+      containers:
+        - command:
```

Comment:
Question -- what is the behavior of this daemonset when the main container completes? Do we need to `sleep infinity` after updating the gpu count label?

Reply:
The Pod is deleted because the node selector no longer applies because of the label overwrite.

Reply:
Ah, I completely overlooked that. Even better.